libraries/3rdparty/CMakeLists.txt

```diff
@@ -1,63 +1,28 @@
-        # Attempt to clone CMock.
-        if( ${BUILD_CLONE_SUBMODULES} )
-            find_package( Git REQUIRED )
-
-            message( "Cloning submodule CMock." )
-            execute_process( COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive libraries/3rdparty/CMock
-                             WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-                             RESULT_VARIABLE CMOCK_CLONE_RESULT )
-
-            if( NOT ${CMOCK_CLONE_RESULT} STREQUAL "0" )
-                message( FATAL_ERROR "Failed to clone CMock submodule." )
-            endif()
-        else()
-            message( FATAL_ERROR "The required submodule CMock does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
-        endif()
-    endif()
-
-    # Build Configuration for CMock and Unity libraries.
-    include_directories("${3RDPARTY_DIR}/CMock/vendor/unity/src/"
-                        "${3RDPARTY_DIR}/CMock/vendor/unity/extras/fixture/src"
-                        "${3RDPARTY_DIR}/CMock/vendor/unity/extras/memory/src"
-                        "${3RDPARTY_DIR}/CMock/src"
-            )
-    link_directories("${CMAKE_BINARY_DIR}/lib"
-            )
-
-    add_library(cmock STATIC
-            "${ROOT_DIR}/libraries/3rdparty/CMock/src/cmock.c"
-            )
-
-    set_target_properties(cmock PROPERTIES
-            ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
-            POSITION_INDEPENDENT_CODE ON
-            COMPILE_FLAGS "-Og"
-            )
-
-    add_library(unity STATIC
-            "${3RDPARTY_DIR}/CMock/vendor/unity/src/unity.c"
-            "${3RDPARTY_DIR}/CMock/vendor/unity/extras/fixture/src/unity_fixture.c"
-            "${3RDPARTY_DIR}/CMock/vendor/unity/extras/memory/src/unity_memory.c"
-            )
-    set_target_properties(unity PROPERTIES
-            ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
-            POSITION_INDEPENDENT_CODE ON
-            )
-
-    target_include_directories(cmock PUBLIC
-            ${ROOT_DIR}/libraries/3rdparty/CMock/src
-            ${ROOT_DIR}/libraries/3rdparty/CMock/vendor/unity/src/
-            ${ROOT_DIR}/libraries/3rdparty/CMock/examples
-            )
-    target_link_libraries(cmock unity)
-
-    add_custom_target(coverage
-            COMMAND ${CMAKE_COMMAND} -P ${CMAKE_SOURCE_DIR}/tools/cmock/coverage.cmake
-            DEPENDS cmock unity http_utest mqtt_utest
-            WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-            )
-endif()
\ No newline at end of file
+# Build Configuration for CMock and Unity libraries.
+add_library(cmock STATIC
+        "${ROOT_DIR}/libraries/3rdparty/CMock/src/cmock.c"
+        )
+
+set_target_properties(cmock PROPERTIES
+        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        POSITION_INDEPENDENT_CODE ON
+        COMPILE_FLAGS "-Og"
+        )
+
+add_library(unity STATIC
+        "${3RDPARTY_DIR}/CMock/vendor/unity/src/unity.c"
```

Comment:
Should we externalize unity and cmock paths in `unity_filepaths.cmake` and `cmock_filepaths.cmake` and include them, the way it is done for HTTP file paths.

Reply:
The utility of separating out the HTTP and MQTT file paths in their individual `filePaths.cmake` is their use in building the library targets (`mqtt`, `http`) and their tests targets (`mqtt_utest`, `http_utest`), both the categories that need lists of the sources and the header files.. It is not the same for Unity/CMock as we only need their library targets; we are not testing them.